crates/starknet-devnet-server/src/restrictive_methods.rs

```diff
@@ -0,0 +1,119 @@
+use std::collections::HashMap;
+
+use lazy_static::lazy_static;
+lazy_static! {
+    static ref MAPPING_HTTP_URIS_TO_RPC_METHODS: HashMap<&'static str, &'static str> =
+        HashMap::from([
+            ("/dump", "devnet_dump"),
+            ("/load", "devnet_load"),
+            ("/set_time", "devnet_setTime"),
+            ("/increase_time", "devnet_increaseTime"),
+            ("/create_block", "devnet_createBlock"),
+            ("/abort_blocks", "devnet_abortBlocks"),
+            ("/restart", "devnet_restart"),
+            ("/mint", "devnet_mint"),
+            ("/postman/load_l1_messaging_contract", "devnet_postmanLoad"),
+            ("/postman/flush", "devnet_postmanFlush"),
+            ("/postman/send_message_to_l2", "devnet_postmanSendMessageToL2"),
+            ("/postman/consume_message_from_l2", "devnet_postmanConsumeMessageFromL2"),
+            ("/predeployed_accounts", "devnet_getPredeployedAccounts"),
+            ("/account_balance", "devnet_getAccountBalance"),
+            ("/config", "devnet_getConfig"),
+        ]);
+    static ref MAPPING_RPC_METHODS_TO_HTTP_URIS: HashMap<&'static str, &'static str> =
+        MAPPING_HTTP_URIS_TO_RPC_METHODS.iter().map(|(k, v)| (*v, *k)).collect();
+    pub static ref DEFAULT_RESTRICTED_JSON_RPC_METHODS: Vec<&'static str> = vec![
+        "devnet_mint",
+        "devnet_restart",
+        "devnet_createBlock",
+        "devnet_abortBlocks",
+        "devnet_impersonateAccount",
+        "devnet_autoImpersonate",
+        "devnet_getPredeployedAccounts"
+    ];
+    static ref DEFAULT_RESTRICTED_HTTP_URIS: Vec<&'static str> =
+        DEFAULT_RESTRICTED_JSON_RPC_METHODS
+            .iter()
+            .filter_map(|method| MAPPING_RPC_METHODS_TO_HTTP_URIS.get(method))
+            .copied()
+            .collect();
+}
+
+pub(crate) fn is_json_rpc_method_restricted(
+    json_rpc_method: &str,
+    restricted_methods: &[&str],
+) -> bool {
+    if restricted_methods.contains(&json_rpc_method) {
+        return true;
+    }
+
+    match MAPPING_RPC_METHODS_TO_HTTP_URIS.get(json_rpc_method) {
+        Some(http_uri) => restricted_methods.contains(http_uri),
+        None => false,
+    }
+}
+
+pub(crate) fn is_uri_path_restricted(uri_path: &str, restricted_uris: &[&str]) -> bool {
+    if restricted_uris.contains(&uri_path) {
+        return true;
+    }
+
+    match MAPPING_HTTP_URIS_TO_RPC_METHODS.get(uri_path) {
+        Some(json_rpc_method) => restricted_uris.contains(json_rpc_method),
+        None => false,
+    }
```

Comment:
Something looks wrong in these two methods. Why are we checking if `http_uri` is contained in `restricted_methods`? Shouldn't that always be false?

Is this just to allow users to specify methods as well as uri paths on startup?

Reply:
So as suggested in the last question of my previous comment, it is to allow specifying methods and URI paths. But it still looks wrong; it means that a container called `restricted_uris` contains rpc method names AND http uris.

Reply:
yes, its possible. User specifies whatever he wants.